### PR TITLE
feat: search feature added and some links have given

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR - 
1. Search feature is added.
2. select from the category is been added.
3. some routes have been added.

please review and provide feedback

### CODE REVIEW

File: src/reducerFunctions/filterReducer/reducerUtilities/filterSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+export const filterBySearch = (productData, searchItem) => {
+    if (searchItem) {
+        return productData.filter(productItem =>
+            productItem.name.toLowerCase().includes(searchItem.toLowerCase()));
+    }
+    return productData;
+
+}
```
Comment (mariyasada) [inline src/reducerFunctions/filterReducer/reducerUtilities/filterSearch.jsx:9]: liked that you converted the query into lowercase

**Review by mariyasada — APPROVED** [review]
this looks fine to me.good to merge

File: src/reducerFunctions/filterReducer/reducerUtilities/filterSearch.jsx (hunk 1 above)
Comment (ShreyashMahajan) [reply src/reducerFunctions/filterReducer/reducerUtilities/filterSearch.jsx:9]: thank you

File: src/pages/homePage/home.jsx [hunk 2]
```diff
@@ -160,12 +161,11 @@ export const Home = () => {
                             </div>
                             <p className="product--downpayment">Zero downpayment</p>
                             <div className="product-cta">
-                                <a href="/cart_page/cart_page.html"> <button className="product__btn">Buy Now</button> </a>
                                 <img src={wishlist} alt="" className="product-wishlist" />
                             </div>
                         </div>
                     </div>
-                    <a href="/product_listing/product_listing.html"> <button className="btn-view-all">View all</button> </a>
+                    <NavLink to='/productListing'><button className="btn-view-all">View all</button> </NavLink>
```
Comment (JantuDeb) [inline src/pages/homePage/home.jsx:168]: Use Link for simple link, if it is navigation link then NAVLink

File: src/context/filterContext/filterContext.jsx [hunk 3]
```diff
@@ -7,20 +7,21 @@ import { sortDataByRange } from '../../reducerFunctions/filterReducer/reducerUti
...
 
 const FilterProvider = ({ children }) => {
     const { productList } = useProduct();
-    const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
+    const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortBySearch: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
 
     const sortedData = sortDataByRange(productList, state.sortbyrange);
     const filteredData = filterByRating(sortedData, state.sortByRating);
     const filterDataByPrice = filterByPrice(filteredData, state.priceValue);
     const dataByCategory = filterDataByCategory(filterDataByPrice, state.category.isBike, state.category.isScooter);
```
Comment (sushil-mod) [inline src/context/filterContext/filterContext.jsx:22]: this is also correct , just a suggestion if you have time you can try filter chaining or compose

**Review by JantuDeb — APPROVED** [review]
Added a suggestion, Liked that you implemented search

**Review by sushil-mod — APPROVED** [review]
just a suggestion , overall code is fine and in UI category sort from home also working fine

File: src/context/filterContext/filterContext.jsx (hunk 3 above)
Comment (ShreyashMahajan) [reply src/context/filterContext/filterContext.jsx:22]: thank you, i will do change that

File: src/pages/homePage/home.jsx (hunk 2 above)
Comment (ShreyashMahajan) [reply src/pages/homePage/home.jsx:168]: thank you, actually that line is deleted

**Pull request merged**